packages/camera/camera/example/ios/RunnerTests/FLTCamPhotoCaptureTests.m

```diff
@@ -0,0 +1,172 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@import camera;
+@import camera.Test;
+@import AVFoundation;
+@import XCTest;
+#import <OCMock/OCMock.h>
+
+@interface FLTCamPhotoCaptureTests : XCTestCase
+
+@end
+
+@implementation FLTCamPhotoCaptureTests
+
+- (void)testCaptureToFile_savePhotoDelegateReferencesMustBeAccessedOnCaptureSessionQueue {
+  XCTestExpectation *setReferenceExpectation =
+      [self expectationWithDescription:
+                @"FLTSavePhotoDelegate references must be set on capture session queue."];
+  XCTestExpectation *clearReferenceExpectation =
+      [self expectationWithDescription:
+                @"FLTSavePhotoDelegate references must be cleared on capture session queue."];
+
+  dispatch_queue_t captureSessionQueue = dispatch_queue_create("capture_session_queue", NULL);
+  const char *captureSessionQueueSpecific = "capture_session_queue";
+  dispatch_queue_set_specific(captureSessionQueue, captureSessionQueueSpecific,
+                              (void *)captureSessionQueueSpecific, NULL);
+  FLTCam *cam = [self createFLTCamWithCaptureSessionQueue:captureSessionQueue];
+
+  // settings.uniqueID is used as the key for `inProgressSavePhotoDelegates` dictionary
+  AVCapturePhotoSettings *settings = [AVCapturePhotoSettings photoSettings];
+  id mockSettings = OCMClassMock([AVCapturePhotoSettings class]);
+  OCMStub([mockSettings photoSettings]).andReturn(settings);
+
+  // We need to make sure the delegate reference is actually saved in a real dictionary, so that we
+  // can call its completion handler later. Must use a partial mock, in order to forward invocation
+  // to the real object.
+  id mockDelegates = OCMPartialMock([NSMutableDictionary dictionary]);
+  OCMStub([mockDelegates setObject:OCMOCK_ANY forKeyedSubscript:OCMOCK_ANY])
```

Comment:
This seems like it's mocking too much of the underlying implementation.  Can you change it to validate a higher level: that when the photo is successfully taken, success or failure message is sent back to the engine?

Reply:
This test is to verify that references are accessed on the right queue. The success/failure tests are the next 2 tests below.